mgmtd/src/lib.rs

```diff
@@ -152,6 +164,27 @@ pub async fn start(info: StaticInfo, license: LicenseVerifier) -> Result<RunCont
     })
 }
 
+// Db schema auto upgrade. This requires slightly different handling and logging than the version
+// in main.rs.
+async fn upgrade_db(db: &sqlite::Connections) -> Result<()> {
+    db.conn(|conn| {
+        let backup_file = sqlite::backup_db(conn)?;
+        log::warn!("Old database backed up to {backup_file:?}");
+        Ok(())
+    })
+    .await?;
+
+    let version = db
+        .write_tx(|tx| {
+            sqlite::migrate_schema(tx, db::MIGRATIONS)
+                .with_context(|| "Upgrading database schema failed")
+        })
+        .await?;
+
+    log::warn!("Database upgraded to version {version}");
```

Comment:
suggestion(non-blocking): 

```suggestion
    log::warn!("Database automatically upgraded to version {version}");
```